cranelift/wasm/src/environ/spec.rs

```diff
@@ -21,13 +21,73 @@ use std::boxed::Box;
 use thiserror::Error;
 use wasmparser::BinaryReaderError;
 use wasmparser::Operator;
+#[cfg(feature = "enable-serde")]
+use serde::{Deserialize, Serialize};
+
+/// Equivalent of `wasmparser`'s FuncType.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
+pub struct WasmFuncType {
+    /// Function params types.
+    pub params: Box<[WasmType]>,
```

Comment:
Not really something for this PR, but I kind of wish we had a `wasm-types` crate or something that our whole crate graph could depend on, so that we weren't converting back and forth between identical types all the time.